Walking the MIB LogStash::SnmpClientError: unknown variable syntax

So being a little lazy lazy, I just wanted to walk all of the .1 tree

```
input{
snmp {
    oid_root_skip => 0
    interval => 30
    walk => [".1"]
    mib_paths => [ "/etc/logstash/rhel6_mibs_dics" ]
    hosts => [
        {
         host => "udp:127.0.0.1/161"
         community => "public"
        }
        ]
    add_field => { "dst_index" => "snmp" }
}
}
```

And it produces an error
```
[INFO ] 2018-09-08 12:12:25.497 [Converge PipelineAction::Create<main>] pipeline - Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>48, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>50}
[INFO ] 2018-09-08 12:12:28.137 [Converge PipelineAction::Create<main>] pipeline - Pipeline started successfully {:pipeline_id=>"main", :thread=>"#<Thread:0x52619d0e@/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:245 sleep>"}
[INFO ] 2018-09-08 12:12:28.165 [Ruby-0-Thread-1: /usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/stud-0.0.23/lib/stud/task.rb:22] agent - Pipelines running {:count=>1, :running_pipelines=>[:main], :non_running_pipelines=>[]}
[INFO ] 2018-09-08 12:12:28.251 [Api Webserver] agent - Successfully started Logstash API endpoint {:port=>9601}
[ERROR] 2018-09-08 12:12:35.976 [[main]<snmp] snmp - error invoking walk operation on OID: 1, ignoring {:exception=>#<LogStash::SnmpClientError: unknown variable syntax 68, Opaque>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp/client.rb:128:in `coerce'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp/client.rb:96:in `block in walk'", "org/jruby/java/proxies/ArrayJavaProxy.java:450:in `each'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp/client.rb:91:in `block in walk'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp/client.rb:80:in `walk'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp.rb:118:in `block in run'", "org/jruby/RubyArray.java:1734:in `each'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp.rb:116:in `block in run'", "org/jruby/RubyArray.java:1734:in `each'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-input-snmp-0.1.0.beta4/lib/logstash/inputs/snmp.rb:106:in `run'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:512:in `inputworker'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:505:in `block in start_input'"]}
```

doing the same things with the CLI utility it works fine
```
[root@xxxxxxx pipeline-snmp]# snmpwalk -v2c -c public localhost .1 2>&1 | more
SNMPv2-MIB::sysDescr.0 = STRING: Linux hd1pbk01lx 2.6.32-573.22.1.el6.x86_64 #1 SMP Thu Mar 17 03:23:39 EDT 2016 x86_64
SNMPv2-MIB::sysObjectID.0 = OID: NET-SNMP-MIB::netSnmpAgentOIDs.10
DISMAN-EVENT-MIB::sysUpTimeInstance = Timeticks: (3819827132) 442 days, 2:37:51.32
SNMPv2-MIB::sysContact.0 = STRING: XXX XX XXXX  <XXXXXXXX>
SNMPv2-MIB::sysName.0 = STRING: XXXXXX
SNMPv2-MIB::sysLocation.0 = STRING: Linux
SNMPv2-MIB::sysORLastChange.0 = Timeticks: (8) 0:00:00.08
SNMPv2-MIB::sysORID.1 = OID: SNMP-MPD-MIB::snmpMPDMIBObjects.3.1.1
SNMPv2-MIB::sysORID.2 = OID: SNMP-USER-BASED-SM-MIB::usmMIBCompliance
SNMPv2-MIB::sysORID.3 = OID: SNMP-FRAMEWORK-MIB::snmpFrameworkMIBCompliance
SNMPv2-MIB::sysORID.4 = OID: SNMPv2-MIB::snmpMIB
SNMPv2-MIB::sysORID.5 = OID: TCP-MIB::tcpMIB
SNMPv2-MIB::sysORID.6 = OID: IP-MIB::ip
SNMPv2-MIB::sysORID.7 = OID: UDP-MIB::udpMIB
SNMPv2-MIB::sysORID.8 = OID: SNMP-VIEW-BASED-ACM-MIB::vacmBasicGroup
SNMPv2-MIB::sysORDescr.1 = STRING: The MIB for Message Processing and Dispatching.
SNMPv2-MIB::sysORDescr.2 = STRING: The MIB for Message Processing and Dispatching.
SNMPv2-MIB::sysORDescr.3 = STRING: The SNMP Management Architecture MIB.
SNMPv2-MIB::sysORDescr.4 = STRING: The MIB module for SNMPv2 entities
SNMPv2-MIB::sysORDescr.5 = STRING: The MIB module for managing TCP implementations
SNMPv2-MIB::sysORDescr.6 = STRING: The MIB module for managing IP and ICMP implementations
SNMPv2-MIB::sysORDescr.7 = STRING: The MIB module for managing UDP implementations
SNMPv2-MIB::sysORDescr.8 = STRING: View-based Access Control Model for SNMP.
SNMPv2-MIB::sysORUpTime.1 = Timeticks: (8) 0:00:00.08
SNMPv2-MIB::sysORUpTime.2 = Timeticks: (8) 0:00:00.08
SNMPv2-MIB::sysORUpTime.3 = Timeticks: (8) 0:00:00.08
```

and if I specify deeper in the mib it works fine

```
input{
snmp {
    oid_root_skip => 0
    interval => 30
    walk => [".1.3.6.1.2.1"]
    mib_paths => [ "/etc/logstash/rhel6_mibs_dics" ]
    hosts => [
        {
         host => "udp:127.0.0.1/161"
         community => "public"
        }
        ]
    add_field => { "dst_index" => "snmp" }
}
}
```
But going deeper in the tree, there is no error. 

![image](https://user-images.githubusercontent.com/6684633/45256570-2e529400-b366-11e8-86de-4389cb1ac4f0.png)

--- a/lib/logstash/inputs/snmp/base_client.rb
+++ b/lib/logstash/inputs/snmp/base_client.rb
@@ -122,6 +122,8 @@ module LogStash
           variable.toString
         when BER::NULL
           NULL_STRING
+        when BER::OPAQUE
+          variable.toString
         when BER::NOSUCHOBJECT
           "Error: No Such Instance currently exists at this OID"
         else

